### PR TITLE
Update RTL13 for 0.9

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

File: Source/ARTRealtimeChannel.m [hunk 1]
```diff
@@ -321,6 +321,10 @@ - (void)transition:(ARTRealtimeChannelState)state status:(ARTStatus *)status {
         [_attachedEventEmitter emit:[NSNull null] with:[ARTErrorInfo createWithCode:90000 message:msg]];
     }
 
+    [self unlessStateChangesBefore:self.realtime.options.channelRetryTimeout do:^{
```
Comment (tcard) [inline Source/ARTRealtimeChannel.m]: You're unconditionally re-attaching here after every channel transition? This has nothing to do with the logic specified at RTL13a and RTL13b.

File: Spec/RealtimeClientChannel.swift [hunk 2]
```diff
@@ -2325,6 +2325,203 @@ class RealtimeClientChannel: QuickSpec {
...
+                            }
+
+                            transport.receive(detachedMessageWithError)
+                        }
+                        
+                        expect(transport.protocolMessagesSent.filter{ $0.action == .Attach }).to(haveCount(2))
+                    }
+
+                    // RTL13b
+                    it("if the attempt to re-attach fails, or if the channel was already in the ATTACHING state, the channel will transition to the SUSPENDED state and the error will be emitted in the ChannelStateChange event") {
```
Comment (tcard) [inline Spec/RealtimeClientChannel.swift]: "if the channel was already in the ATTACHING state" This is not tested.

File: Spec/RealtimeClientChannel.swift [hunk 3]
```diff
@@ -2325,6 +2325,203 @@ class RealtimeClientChannel: QuickSpec {
...
+                                    fail("Reason error is nil"); done(); return
+                                }
+                                expect(stateChange.reason).to(beIdenticalTo(detachedMessageWithError.error))
+                                expect(channel.errorReason).to(beIdenticalTo(detachedMessageWithError.error))
+                                done()
+                            }
+                        }
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.once(.Attaching) { _ in
```
Comment (tcard) [inline Spec/RealtimeClientChannel.swift]: You should check that the wait is close to `channelRetryTimeout`.

Comment (ricardopereira) [inline Spec/RealtimeClientChannel.swift]: Done 5c57381.

**Comment by ricardopereira** [comment]
Thanks, it's passing. Will squash and merge.

**Pull request merged**